src/neat/neat_structures.py

```diff
@@ -36,6 +39,13 @@ def __str__(self):
                f'and is {active_stat}.'
 
 
+class NodeGene:
```

Comment:
There's no real need for a NodeGene class. Can just do self.nodes inside genome class and create a set with each node in it. There's never a new node created without a connection, so it's all handled in just the classes we have.